File: Services/DB_Handler/models/URLComment.py
from sqlalchemy import Column, Integer, DateTime, ForeignKey, Text, String
from .base import Base


class URLComment(Base):
    __tablename__ = 'url_comment'

    comment_id = Column(Integer, primary_key=True, nullable=False)
    url_id = Column(Integer, ForeignKey("url_storage.url_id"), nullable=False)
    writer_id = Column(Integer, nullable=False)
    writer_role = Column(String, nullable=False)
    root_id = Column(Integer, ForeignKey("url_comment.comment_id", ondelete="CASCADE"))
    rating = Column(Integer)
    comment_content = Column(Text, nullable=False)
    created_at = Column(DateTime, nullable=False)

File: Services/Public/Public.Domain/UrlComment.cs
using System.ComponentModel.DataAnnotations;

namespace Public.Domain;

public class UrlComment
{
    [Key] public int comment_id { get; set; }

    [Required] public int url_id { get; set; }

    [Required] public int writer_id { get; set; }
    [Required] public string writer_role { get; set; }

    public int? root_id { get; set; }

    [Required] public int rating { get; set; }

    [Required] public string comment_content { get; set; }

    [Required] public DateTime created_at { get; set; }
}